### PR TITLE
Add Multi-objective to recipe (tutorial)

### PR DESCRIPTION
**Pull request by crcrpar**
4599a30 is based on https://github.com/optuna/optuna/issues/1721#issuecomment-767970082.

related #1721

### CODE REVIEW

**Comment by hvy** [comment]
@himkt could you jump in as a reviewer here?

File: tutorial/20_recipes/002_multi_objective.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+"""
+.. _multi_objective_lgbm:
+
+Multi-objective optimization example with LightGBM
+==================================================
+
+This tutorial is an example of multi-objective optimization with Optuna which optimizes
+precision and recall while it would be more common to use F-measure and (ROC-)AUC instead.
+
+As of `v2.4.0 <https://github.com/optuna/optuna/releases/tag/v2.4.0>`_, :class:`~optuna.multi_objective.MultiObjectiveStudy` is merged into :class:`~optuna.study.Study`.
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:10]:
```suggestion
Starting from `v2.4.0 <https://github.com/optuna/optuna/releases/tag/v2.4.0>`_, :class:`~optuna.multi_objective.MultiObjectiveStudy` is made available with :class:`~optuna.study.Study`.
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+"""
+.. _multi_objective_lgbm:
+
+Multi-objective optimization example with LightGBM
+==================================================
+
+This tutorial is an example of multi-objective optimization with Optuna which optimizes
+precision and recall while it would be more common to use F-measure and (ROC-)AUC instead.
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:8]:
```suggestion
precision and recall. In practice, it is more common to use F-measure and (ROC-)AUC.
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import numpy
+from sklearn import datasets
+from sklearn.metrics import precision_score, recall_score
+from sklearn.model_selection import train_test_split
+
+import optuna
+
+
+###################################################################################################
+# In the ``objective`` below, we train a model using Breast cancer dataset before
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:26]:
```suggestion
# In the ``objective`` below, we train a model using the Breast cancer dataset before
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    preds = gbm.predict(valid_x)
+    pred_labels = numpy.rint(preds)
+    precision = precision_score(valid_y, pred_labels)
+    recall = recall_score(valid_y, pred_labels)
+    return precision, recall
+
+
+###################################################################################################
+# To make :class:`~optuna.study.Study` compatible with multi-objective optimization,
+# what we need to do is only to pass the collection of ``"direction"``\\s.
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:57]: Seems like the backslash isn't escaped properly. Could you have a look? https://65701-122299416-gh.circle-artifacts.com/0/docs/build/html/tutorial/20_recipes/002_multi_objective.html#sphx-glr-tutorial-20-recipes-002-multi-objective-py

File: tutorial/20_recipes/002_multi_objective.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+optuna.logging.get_logger("optuna").addHandler(logging.StreamHandler(sys.stdout))
+
+study = optuna.create_study(directions=("maximize", "maximize"))
+study.optimize(objective, n_trials=100)
+
+
+###################################################################################################
+# Optuna provides a visualization feature unique to multi-objective: :func:`~optuna.visualization.plot_pareto_front`.
+optuna.visualization.plot_pareto_front(study)
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:71]: Just an idea but how about passing `target_names` with "precision" and "recall"? That might be a tiny bit more pedagogical.

File: tutorial/20_recipes/002_multi_objective.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
+"""
+.. _multi_objective_lgbm:
+
+Multi-objective optimization example with LightGBM
```
Comment (hvy) [inline tutorial/20_recipes/002_multi_objective.py:4]: A nit (for consistency).
```suggestion
Multi-objective Optimization Example with LightGBM
```

**Review by hvy — CHANGES_REQUESTED** [review]
Thanks for the great PR. It's very simple and I think it'd help users to quickly get started with MO. The changes mostly LGTM so I left smaller comments, mostly on the English phrasings.

I by the way think that it could be even further improved it we'd be explicit with the sampler and explain that some support MO while some don't. Maybe it's beyond the scope of this tutorial? Anyhow, I think it can be left outside this PR.

**Comment by toshihikoyanase** [comment]
Just FYI: I update the example notebook using the UCI Bank Marketing dataset. Please refer to https://github.com/optuna/optuna/issues/1721#issuecomment-769759092.

File: tutorial/20_recipes/002_multi_objective.py (hunk 4 above)
Comment (HideakiImamura) [reply tutorial/20_recipes/002_multi_objective.py:57]: I think we just need to remove the backslash here.
```suggestion
# what we need to do is only to pass the collection of ``"direction"``s.
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. The content is interesting and easy to follow. How about using the UCI Bank dataset as @toshihikoyanase suggested?

**Comment by toshihikoyanase** [comment]
@HideakiImamura Thank you for your suggestion. I talked about changing the target library from lightgbm to PyTorch with @crcrpar. The corresponding PR is #2305. Readers may understand the PyTorch's scenario that optimizes flops and accuracy more easily than the current precision-recall optimization.

It will double the build time (3min to 6min in GitHub Actions), but it seems to keep the build time limitation of ReadTheDocs.

**Comment by HideakiImamura** [comment]
Thanks for the explanation. So, should we close this PR?

**Comment by toshihikoyanase** [comment]
I think so, but I'd like to wait for the author's response.
